Is xml-rs memory leak malloc_trim workaround still relevant?

https://github.com/pop-os/cosmic-store/blob/1e1b322/src/backend/mod.rs#L91-L99

Doesn't build if host system is using e.g. https://musl.libc.org:
```
   Compiling cosmic-store v0.1.0 (.../cosmic-epoch/cosmic-store)
error[E0425]: cannot find function `malloc_trim` in crate `libc`
  --> src/backend/mod.rs:95:19
   |
95 |             libc::malloc_trim(0);
   |                   ^^^^^^^^^^^ not found in `libc`

For more information about this error, try `rustc --explain E0425`.
error: could not compile `cosmic-store` (bin "cosmic-store") due to 1 previous error
error: Recipe `build-debug` failed on line 41 with exit code 101
error: Recipe `build` failed on line 23 with exit code 101
```

--- a/src/backend/mod.rs
+++ b/src/backend/mod.rs
@@ -108,7 +108,7 @@ pub fn backends(locale: &str, refresh: bool) -> Backends {
     });
 
     //TODO: Workaround for xml-rs memory leak when loading appstream data
-    #[cfg(target_os = "linux")]
+    #[cfg(all(target_os = "linux", target_env = "gnu"))]
     {
         let start = Instant::now();
         unsafe {

